main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -824,6 +824,96 @@ object Typer extends Phase[ResolvedAst.Program, TypedAst.Root] {
             resultType <- unifyM(tvar, Type.Unit, loc)
           } yield resultType
 
+        /*
+         * NewChannel expression.
+         */
+        case ResolvedAst.Expression.NewChannel(exp, tpe, loc) =>
+          //
+          //  e: Int8 | Int16 | Int32 | Int64 | BigInt
+          //  ----------------------------------------
+          //  channel t e : Channel[t]
+          //
+          for {
+            texp <- visitExp(exp)
+            _ <- texp match {
+              case Type.Int8 => unifyM(texp, Type.Int8, loc)
+              case Type.Int16 => unifyM(texp, Type.Int16, loc)
+              case Type.Int32 => unifyM(texp, Type.Int32, loc)
+              case Type.Int64 => unifyM(texp, Type.Int64, loc)
+              case Type.BigInt => unifyM(texp, Type.BigInt, loc)
+              case _ => unifyM(texp, Type.Int32, loc)
+            }
+              
+            resultType <- liftM(Type.mkChannel(tpe))
+          } yield resultType
+
+        /*
+         * GetChannel expression.
+         */
+        case ResolvedAst.Expression.GetChannel(exp, tvar, loc) =>
+          // exp : Channel[t]
+          // ----------------
+          // <- exp : t
+          for (
+            tpe  <- visitExp(exp);
```

Comment:
there is a missing requirement that the type of exp should be Channel[t] for some t.